.github/workflows/tests.yml

```diff
@@ -1,22 +1,57 @@
-name: Tests
-on: push
+name: CI
+
+permissions:
+  pull-requests: write
+  contents: write
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+env:
+  RUSTFLAGS: -D warnings
+  RUST_BACKTRACE: short
+  CARGO_NET_RETRY: 10
+  CARGO_TERM_COLOR: always
+  CARGO_INCREMENTAL: 0
+
 jobs:
-  workflows:
-    strategy:
-      matrix:
-        platform: [ubuntu-latest, macos-latest]
-    runs-on: ${{ matrix.platform }}
+  test:
+    runs-on: ubuntu-20.04
+
     steps:
-      - name: Checkout branch
-        uses: actions/checkout@v2
-      - name: Install Node
-        uses: actions/setup-node@v2
-      - name: Install Rust
-        uses: actions-rs/toolchain@v1
-        with:
-          toolchain: stable
-          target: wasm32-unknown-unknown
-      - name: Install modules
-        run: yarn
-      - name: Run tests
-        run: yarn test
```

Comment:
no need to install rust, you can simply call `cargo test`, see https://github.com/near-examples/hello-near-examples/blob/main/.github/workflows/tests-rs.yml